packages/phone-number-privacy/common/test/utils/sequential-delay.test.ts

```diff
@@ -0,0 +1,268 @@
+import { SequentialDelayDomain } from '../../src/interfaces'
+import { checkSequentialDelay, SequentialDelayResult } from '../../src/utils/sequential-delay'
+
+type TestAttempt = {
+  timestamp: number
+  expectedResult: SequentialDelayResult
+}
+
+describe('Sequential Delay Test Suite', () => {
+  const checkTestAttempts = (t: number, domain: SequentialDelayDomain, attempts: TestAttempt[]) => {
+    let result: SequentialDelayResult | undefined
+    attempts.forEach((attempt) => {
```

Comment:
I've always been told, and I agree, to prefer `for ... of` to `forEach`
```suggestion
    for (const attempt of attempts) {
```